File: syntagNet/src/main/java/org/sqlunet/syntagnet/sql/CollocationQueryFromWordId.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.sqlunet.syntagnet.sql;

import android.database.sqlite.SQLiteDatabase;

/**
 * SyntagNet Collocation query from word id
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
class CollocationQueryFromWordId extends BaseCollocationQuery
{
	/**
	 * <code>QUERY</code> is the SQL statement
	 */
	static private final String QUERY = SqLiteDialect.SyntagNetCollocationQueryFromWordId;

	/**
	 * Constructor
	 *
	 * @param connection connection
	 * @param wordid     target word id
	 */
	@SuppressWarnings("boxing")
	public CollocationQueryFromWordId(final SQLiteDatabase connection, final long wordid)
	{
		super(connection, CollocationQueryFromWordId.QUERY);
		setParams(wordid, wordid);
	}
}

File: syntagNet/src/main/java/org/sqlunet/syntagnet/sql/CollocationQueryFromWordId.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.syntagnet.sql

import android.database.sqlite.SQLiteDatabase

/**
 * SyntagNet Collocation query from word id
 *
 * @param connection connection
 * @param wordid     target word id
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
internal class CollocationQueryFromWordId(connection: SQLiteDatabase, wordid: Long) : BaseCollocationQuery(connection, QUERY) {

    init {
        setParams(wordid, wordid)
    }

    companion object {
        /**
         * `QUERY` is the SQL statement
         */
        private const val QUERY = SqLiteDialect.SyntagNetCollocationQueryFromWordId
    }
}
